src/py/flwr/client/app.py

```diff
@@ -319,7 +319,7 @@ def _on_backoff(retry_state: RetryState) -> None:
         on_backoff=_on_backoff,
     )
 
-    node_state = NodeState(partition_id=partition_id)
+    node_state = NodeState(node_id=-1, node_config={}, partition_id=partition_id)
```

Comment:
`-1` is a possible `node_id`. we might need to find a different solution here. with `grpc-rere` and `rest`, the `node_id` should not change and we should not call `create_node()` on subsequent connects. we can lazily instantiate `NodeState` the first time `create_node` gets called, and then call `create_node` never again. the fact that `NodeState` was initialized can actually act as a flag to determine whether we should call `create_node`.